Migrate UI register catalyst

folarinemurgo opened the pull request:
## Description / Change(s) / Related issue(s)
Updated Catalyst Voting

##### Ticket
YV-483

Opestein committed:
rn79: Migrate UI (Register Catalyst)
Styling and Components Updated

Opestein committed:
rn79: Migrate UI (Register Catalyst)
Styling and Components Updated

Opestein committed:
rn79: Migrate UI (Register Catalyst)
Completed.
Switched crypto-random-string to v5.
switched pbkdf2 to react-native-fast-crypto as pbkdf2 requires metro to use crypto-browserify and stream-browserify polyfills

Opestein committed:
rn79: Migrate UI (Register Catalyst)
Completed.
Switched crypto-random-string to v5.
switched pbkdf2 to react-native-fast-crypto as pbkdf2 requires metro to use crypto-browserify and stream-browserify polyfills

Opestein committed:
rn79: Migrate UI (Register Catalyst)
Completed.
Switched crypto-random-string to v5.
switched pbkdf2 to react-native-fast-crypto as pbkdf2 requires metro to use crypto-browserify and stream-browserify polyfills

stackchain commented on apps/mobile/package.json:
```diff
@@ -75,11 +75,15 @@
     "@yoroi/theme": "file:../../packages/theme",
     "@yoroi/transfer": "file:../../packages/transfer",
     "@yoroi/types": "file:../../packages/types",
+    "assert": "^2.1.0",
```
why?

stackchain commented on apps/mobile/package.json:
```diff
@@ -136,7 +145,10 @@
     "react-native-view-shot": "^4.0.3",
     "react-native-web": "^0.20.0",
     "react-native-webview": "13.13.5",
+    "readable-stream": "^4.7.0",
```
i see a lot of new packages URL is "polyfilled" already

stackchain commented on apps/mobile/package.json:
```diff
@@ -105,6 +109,10 @@
     "immer": "^10.1.1",
     "lodash": "^4.17.21",
     "lottie-react-native": "7.2.2",
+    "os-browserify": "^0.3.0",
```
oO

stackchain commented on apps/mobile/src/wallets/cardano/catalyst/catalystCipher.ts:
```diff
@@ -17,33 +18,27 @@ const PROTO_VERSION = Buffer.from('01', 'hex')
-  const derivedB64 = await Pbkdf2.derive(
-    passwordB64,
-    saltB64,
+export function promisifyPbkdf2(password: Uint8Array, salt: Buffer): Buffer {
```
is it still promisifying? it looks it can be renamed

stackchain replied:
@lisicky do we have it anywhere else yoroi-lib / csl the fucntions for catalyst registration?

Opestein committed:
rn79: Migrate UI (Register Catalyst)
Removed unused packages

folarinemurgo commented:
Resolved @stackchain

banklesss reviewed (changes requested):
hey @folarinemurgo

Could you please revert the review tx changes? those are in develop already

banklesss commented:
Also check out the rest of things that don't depend on catalyst to revert them too pls

folarinemurgo commented:
> hey @folarinemurgo
> 
> Could you please revert the review tx changes? those are in develop already

Hello @banklesss 
I don't have changes on review tx; or maybe i don't get the changes you're referring to

banklesss commented:
> Hello @banklesss
>I don't have changes on review tx; or maybe i don't get the changes you're referring to

Review the changes in the Files tab. I made several changes to your code in the TX review.

banklesss commented:
@folarinemurgo you can use `git checkout origin/develop <path-to-folder>` to revert those changes i mentioned

folarinemurgo commented:
> @folarinemurgo you can use `git checkout origin/develop <path-to-folder>` to revert those changes i mentioned

There were merge conflicts which was i rebased on develop; i didn't touch review tx screens at all

> @folarinemurgo you can use `git checkout origin/develop <path-to-folder>` to revert those changes i mentioned

Reverted merge

stackchain committed:
Merge branch 'develop' into Migrate-UI-Register-Catalyst
